File: contract/src/destroypoll.cpp
#include "contract.hpp"
#include <cmath>
#include <map>

namespace eosstrawpoll
{

void contract::destroypoll(const eosio::name poll_id)
{
    auto poll = polls_table.find(poll_id.value);
    eosio::check(poll != polls_table.end(), "poll doesn't exist");

    require_auth(poll->account);

    polls_table.erase(poll);

    auto popular_poll = popular_table.find(poll_id.value);
    if (popular_poll != popular_table.end())
    {
        popular_table.erase(popular_poll);
    }

    auto new_poll = latest_table.find(poll_id.value);
    if (new_poll != latest_table.end())
    {
        latest_table.erase(new_poll);
    }
}

} // namespace eosstrawpoll

File: contract/src/destroypoll.rs
use crate::types::*;
use eosio::*;

#[eosio_action]
pub fn destroypoll(name: PollName) {
    let code = AccountName::receiver();
    let table = Poll::table(code, code);
    let cursor = table.find(name).assert("poll doesn't exist");

    let poll = cursor.get().assert("read");
    poll.account.require_auth();

    cursor.remove().assert("read");

    // let table = get_popular_table();
    // let itr = table.find(name);
    // if !table.is_end(itr) {
    //     table.erase(itr);
    // }

    // let table = get_new_table();
    // let itr = table.find(name);
    // if !table.is_end(itr) {
    //     table.erase(itr);
    // }
}
